### PR TITLE
Update to use tigerbeetle/main

### PR DESCRIPTION
**Pull request by kprotty**
* Rebuilds pkg native binaries
* Updates C header and Go types / tests
* Submodule points to `tigerbeetle/main-sync` which includes `tb_client` but is rebased onto `main`

### CODE REVIEW

File: pkg/types/main.go [hunk 1]
```diff
@@ -79,113 +87,94 @@ type Transfer struct {
...
-	CommitDebitAccountNotFound        uint32 = 10
-	CommitCreditAccountNotFound       uint32 = 11
-	CommitDebitAmount_wasNotReserved  uint32 = 12
-	CommitCreditAmount_wasNotReserved uint32 = 13
-	CommitExceedsCredits              uint32 = 14
-	CommitExceedsDebits               uint32 = 15
-	CommitConditionRequiresPreimage   uint32 = 16
-	CommitPreimageRequiresCondition   uint32 = 17
-	CommitPreimageInvalid             uint32 = 18
+	TransferLinkedEventFailed                          uint32 = 0
```
Comment (DonChangfoot) [inline pkg/types/main.go:131]: Should this also start at 1?

File: pkg/types/main.go [hunk 2]
```diff
@@ -56,17 +55,26 @@ func (f AccountFlags) ToUint32() uint32 {
...
-	AccountExistsWithDifferentUserData      uint32 = 3
-	AccountExistsWithDifferentReservedField uint32 = 4
-	AccountExistsWithDifferentUnit          uint32 = 5
-	AccountExistsWithDifferentCode          uint32 = 6
-	AccountExistsWithDifferentFlags         uint32 = 7
-	AccountExceedsCredits                   uint32 = 8
-	AccountExceedsDebits                    uint32 = 9
-	AccountReservedField                    uint32 = 10
-	AccountReservedFlagPadding              uint32 = 11
+	AccountLinkedEventFailed                 uint32 = 0
```
[the remark this answers is not in the document]
Comment (kprotty) [reply pkg/types/main.go:58]: Good catch

File: pkg/types/main.go [hunk 3]
```diff
@@ -79,113 +87,94 @@ type Transfer struct {
...
 }
 
 type TransferFlags struct {
-	Linked         bool
-	TwoPhaseCommit bool
-	Condition      bool
+	Linked              bool
+	Pending             bool
+	PostPendingTransfer bool
+	VodiPendingTransfer bool
```
[the remark this answers is not in the document]
Comment (kprotty) [reply pkg/types/main.go:105]: thx

**Review by DonChangfoot — APPROVED** [review]
thanks @kprotty. LGTM!

**Pull request merged**